x-pack/filebeat/input/awss3/input.go

```diff
@@ -46,17 +49,41 @@ func (im *s3InputManager) Create(cfg *conf.C) (v2.Input, error) {
 		return nil, fmt.Errorf("initializing AWS config: %w", err)
 	}
 
-	// The awsConfig now contains the region from the credential profile or default region
-	// if the region is explicitly set in the config, then it wins
-	if config.RegionName != "" {
+	if config.AccessPointARN != "" {
+		// When using the access point ARN, requests must be directed to the
+		// access point hostname. The access point hostname takes the form
+		// AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com
+		arnParts := strings.Split(config.AccessPointARN, ":")
+		region := arnParts[3]
+		accountID := arnParts[4]
+		accessPointName := strings.Split(arnParts[5], "/")[1]
```

Comment:
what if the string is `arn:aws:s3:us-east-1:123456789:accesspoint/`, I don't see `isValidAccessPointARN` method catching this edge case, and even though it will not cause a panic since [1] will be an empty string, I see this could lead of errors down stream if not caught before.

Reply:
Makes sense! I added this edge case to `isValidAccessPointARN`.